# Loading and Preprocessing Dataset

Before training the model, we need to load the dataset and remove the 'Enrolled' students. We are framing this as a binary classification problem to predict strictly between 'Dropout' and 'Graduate'.

In [7]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split
from sklearn.metrics import (
    accuracy_score,
    classification_report,
    confusion_matrix,
    ConfusionMatrixDisplay
)
import xgboost as xgb

# Load the dataset
df = pd.read_csv("../dataset/data.csv", sep=";")

# Remove 'Enrolled' class
df = df[df['Target'] != 'Enrolled']

df.head()

,Marital status,Application mode,Application order,Course,Daytime/evening attendance\t,Previous qualification,Previous qualification (grade),Nacionality,Mother's qualification,Father's qualification,...,Curricular units 2nd sem (credited),Curricular units 2nd sem (enrolled),Curricular units 2nd sem (evaluations),Curricular units 2nd sem (approved),Curricular units 2nd sem (grade),Curricular units 2nd sem (without evaluations),Unemployment rate,Inflation rate,GDP,Target
0,1,17,5,171,1,1,122.0,1,19,12,...,0,0,0,0,0.000000,0,10.8,1.4,1.74,Dropout
1,1,15,1,9254,1,1,160.0,1,1,3,...,0,6,6,6,13.666667,0,13.9,-0.3,0.79,Graduate
2,1,1,5,9070,1,1,122.0,1,37,37,...,0,6,0,0,0.000000,0,10.8,1.4,1.74,Dropout
3,1,17,2,9773,1,1,122.0,1,38,37,...,0,6,10,5,12.400000,0,9.4,-0.8,-3.12,Graduate
4,2,39,1,8014,0,1,100.0,1,37,38,...,0,6,6,6,13.000000,0,13.9,-0.3,0.79,Graduate


We can see the data set loaded successfully


# Encoding Target & Features

Machine learning model text bujhte pare na. Tai Target column-ke LabelEncoder bebohar kore numeric value (0 and 1) te convert korbo. Erpor baki categorical feature-guloke One-Hot Encoding bebohar kore dummy variable-e porinoto korbo.

In [8]:
le = LabelEncoder()
df["Target"] = le.fit_transform(df["Target"])

print("Class mapping:", dict(zip(le.classes_, le.transform(le.classes_))))
print(df["Target"].value_counts())

df = pd.get_dummies(df, drop_first=True)

Class mapping: {'Dropout': np.int64(0), 'Graduate': np.int64(1)}
Target
1    2209
0    1421
Name: count, dtype: int64


We can see that the text-based categories have been successfully converted into numbers, where 'Dropout' is mapped to 0 and 'Graduate' is mapped to 1. The output also shows the distribution of our dataset, confirming there are 2,209 graduates and 1,421 dropouts. Finally, the remaining text features are transformed into numerical dummy variables so the machine learning model can easily process them for training.

# Spliting Features and Target

Now we separate the input features (X) from the target output (y) and split the data into 80% for training and 20% for testing.

In [9]:
X = df.drop("Target", axis=1)
y = df["Target"]

print("Features Shape:", X.shape)
print("Target Shape:", y.shape)

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

print("Training Set:", X_train.shape)
print("Testing Set:", X_test.shape)

Features Shape: (3630, 36)
Target Shape: (3630,)
Training Set: (2904, 36)
Testing Set: (726, 36)


We can see that our dataset is successfully separated into 3,630 input features and target labels. To prepare for machine learning, this data is then divided into two parts, using 80% of the data (2,904 students) to train the model and saving the remaining 20% (726 students) to accurately test its performance later.